ckan/logic/validators.py

```diff
@@ -861,6 +861,18 @@ def email_validator(value, context):
             raise Invalid(_('Email {email} is not a valid format').format(email=value))
     return value
 
+def email_is_unique(key, data, errors, context):
+    '''Validate email is unique'''
+    model = context['model']
+    session = context['session']
+    user = session.query(model.User).filter_by(email=data[key]).first()
```

Comment:
Oh, if the existing user shows up first, then it won't detect any duplicates. I think you need to filter out the user in this query i.e. `.filter(model.User.name != data[('name',)])` rather than in the `if` in the next statement.

Reply:
@davidread , thank you! I've changed the validator logic. But, since the validator used in two cases (registration and update), I've chose another approach.